app/src/main/java/com/andryoga/safebox/ui/view/MainActivity.kt

```diff
@@ -100,6 +113,44 @@ class MainActivity : AppCompatActivity() {
 
         CrashlyticsKeys(this).setDefaultKeys()
         prefetchReviewInfo()
+
+        appUpdate()
+
+    }
+
+    private fun appUpdate(){
+        val appUpdateInfoTask = appUpdateManager.appUpdateInfo
+        appUpdateInfoTask.addOnSuccessListener{ appUpdateInfo ->
+           if ( appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE
+               && appUpdateInfo.isUpdateTypeAllowed(AppUpdateType.FLEXIBLE)){
+               appUpdateManager.startUpdateFlowForResult(appUpdateInfo,AppUpdateType.FLEXIBLE, this, AUTO_UPDATE_REQUEST_CODE)
+               appUpdateManager.registerListener(listener)
+           }
+            else if ( appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE
+               && appUpdateInfo.isUpdateTypeAllowed(AppUpdateType.IMMEDIATE)){
+               appUpdateManager.startUpdateFlowForResult(appUpdateInfo,AppUpdateType.IMMEDIATE, this, AUTO_UPDATE_REQUEST_CODE)
+            }
+        }
+    }
+
+    private val listener = InstallStateUpdatedListener{ state ->
+        if (state.installStatus() == InstallStatus.DOWNLOADED){
+            snackbarForCompleteUpdate()
+        }
+    }
+
+    private fun snackbarForCompleteUpdate(){
+        Snackbar.make(findViewById(R.id.content), "App update has been downloaded", Snackbar.LENGTH_INDEFINITE).
+        setAction("Install"){
```

Comment:
"Install" need to be taken from string resource, should not be hardcoded in code